dap2-service/src/main/scala/latis/service/dap2/DdsEncoder.scala

```
@@ -0,0 +1,12 @@
+package latis.service.dap2
+
+import cats.effect.IO
+import fs2.Stream
+
+import latis.dataset.Dataset
+import latis.output.Encoder
+
+class DdsEncoder extends Encoder[IO, String]{
+  override def encode(dataset: Dataset): Stream[IO, String] =
+    Stream.emit(Dds.fromDataset(dataset).fold(_.message, _.asString))
```

Not sure how exactly we want to handle a possible `LatisException` returned by `fromDataset` here. To me, it seems like we could either throw it or just pass the message along to be displayed...

---

For now, we could raise an error in the `Stream`, but it would be better if we could fail before we start streaming an OK response.

Do we need to do something like scodec where our `Encoder` returns an `Attempt`. Another problem for another day.

---

I think this will do what we want for now:

```scala
Stream.fromEither[IO](Dds.fromDataset(dataset).map(_.asString))
```